internal/integration/rest/utils.go

```diff
@@ -1,38 +1,83 @@
 package rest
 
-import "time"
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
 
 type Retry struct {
 	Attempts int
 	MinSleep,
 	MaxSleep time.Duration
 }
 
-// ExtendProjects adds custom fields and created date to projects
-func ExtendProjects(projects []*Project, projectsOData []*ProjectOData) []*Project {
-	for indexProject, project := range projects {
-		// arrays have to have the same order
-		if len(projectsOData) > indexProject && project.ID == projectsOData[indexProject].ID {
-			addDateAndCustomFields(project, projectsOData[indexProject])
-			continue
-		}
-
-		// if we have wrong order in OData array
-		for _, projectOData := range projectsOData {
-			if project.ID == projectOData.ID {
-				addDateAndCustomFields(project, projectOData)
-				break
-			}
-		}
-	}
-
-	return projects
+// GetFilterForProjects get filter string for projects list
+func GetFilterForProjects(fromDate, teamName, projectIds string) string {
+	if teamName == "" && projectIds == "" {
+		return fmt.Sprintf("CreatedDate gt %s", fromDate)
+	}
+	if teamName == "" {
+		return fmt.Sprintf("CreatedDate gt %s and %s", fromDate, getProjectIdsFilter(projectIds))
+	}
+	if projectIds == "" {
+		return fmt.Sprintf("CreatedDate gt %s and %s", fromDate, getTeamFilter(teamName))
+	}
+	return fmt.Sprintf("CreatedDate gt %s and %s and %s", fromDate, getTeamFilter(teamName),
+		getProjectIdsFilter(projectIds))
+}
+
+// GetFilterForProjectsWithLastScan get filter string for projects list with last scan
+func GetFilterForProjectsWithLastScan(fromDate, teamName, projectIds string) string {
```

Comment:
Please add unit tests to this and other functions in this file

Reply:
I've already added these test cases